ambry-frontend/src/main/java/com.github.ambry.frontend/TtlUpdateHandler.java

```diff
@@ -0,0 +1,261 @@
+     */
+    IdConverterCallback(RestRequest restRequest, RestResponseChannel restResponseChannel, Callback<Void> callback) {
+      this.restRequest = restRequest;
+      this.restResponseChannel = restResponseChannel;
+      this.callback = callback;
+    }
+
+    @Override
+    public void onCompletion(String result, Exception exception) {
+      long processingStartTimeMs = SystemTime.getInstance().milliseconds();
+      metrics.idConverterProcessingTimeInMs.update(processingStartTimeMs - operationStartTimeMs);
+      try {
+        if (exception == null) {
+          BlobId blobId = FrontendUtils.getBlobIdFromString(result, clusterMap);
+          accountAndContainerInjector.injectTargetAccountAndContainerFromBlobId(blobId, restRequest);
+          securityService.postProcessRequest(restRequest,
+              new SecurityPostProcessRequestCallback(restRequest, restResponseChannel, blobId, callback));
+        }
+      } catch (Exception e) {
+        exception = e;
+      } finally {
+        metrics.updateBlobTtlIdConverterCallbackProcessingTimeInMs.update(
+            SystemTime.getInstance().milliseconds() - processingStartTimeMs);
+        if (exception != null) {
+          callback.onCompletion(null, exception);
+        }
+      }
+    }
+  }
+
+  /**
+   * Callback for {@link SecurityService#postProcessRequest(RestRequest, Callback)} that forwards the TTL update
+   * request to the {@link Router} on success.
+   */
+  private class SecurityPostProcessRequestCallback implements Callback<Void> {
+    private final RestRequest restRequest;
+    private final RestResponseChannel restResponseChannel;
+    private final BlobId blobId;
+    private final Callback<Void> callback;
+    private final long operationStartTimeMs = SystemTime.getInstance().milliseconds();
+
+    /**
+     * @param restRequest the {@link RestRequest} received
+     * @param restResponseChannel the {@link RestResponseChannel} to return the response over
+     * @param blobId the {@link BlobId} to operate on
+     * @param callback the {@link Callback} to invoke when the response is ready (or if there is an exception).
+     */
+    SecurityPostProcessRequestCallback(RestRequest restRequest, RestResponseChannel restResponseChannel, BlobId blobId,
+        Callback<Void> callback) {
+      this.restRequest = restRequest;
+      this.restResponseChannel = restResponseChannel;
+      this.blobId = blobId;
+      this.callback = callback;
+    }
+
+    @Override
+    public void onCompletion(Void result, Exception exception) {
+      long processingStartTimeMs = SystemTime.getInstance().milliseconds();
+      metrics.updateBlobTtlSecurityPostProcessRequestTimeInMs.update(processingStartTimeMs - operationStartTimeMs);
+      try {
+        if (exception == null) {
+          String serviceId = RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.SERVICE_ID, true);
```

Comment:
I guess we are requiring service-id here since this is a new API? I think it was optional on delete.

Reply:
Right - I think we should require it in everything but left didn't make that change for now